File: app/src/main/java/com/abhi/lokaldemo/LokalApplication.java
package com.abhi.lokaldemo;

import com.abhi.lokaldemo.di.AppComponent;
import com.abhi.lokaldemo.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

public class LokalApplication extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        AppComponent appComponent = DaggerAppComponent.builder().application(this).build();
        appComponent.inject(this);
        return appComponent;
    }
}

File: app/src/main/java/com/abhi/lokaldemo/LokalApplication.kt
package com.abhi.lokaldemo

import com.abhi.lokaldemo.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

class LokalApplication : DaggerApplication() {


    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        val appComponent = DaggerAppComponent.builder().application(this).build()
        appComponent.inject(this)
        return appComponent
    }
}
